src/components/form/HdInputPassword.vue

```diff
@@ -0,0 +1,92 @@
+<template>
+  <div
+    class="passwordInput"
+  >
+    <HdInput
+    :type="currentType"
+    v-model="currentValue"
+    v-bind="$props"
+    :class="{ 'field--hasControl': this.showVisibilityToggle }"
+    />
+
+    <span v-if="showVisibilityToggle"
+      class="field__visibilityToggle"
+      :class="{'field__visibilityToggle--visible': currentType === 'text'}"
+      @click="togglePasswordVisibility"
+    />
+  </div>
+</template>
+
+<script>
+import HdInput from 'hd-blocks/components/form/HdInput.vue';
+
+export default {
+  name: 'HdInputPassword',
+  components: {
+    HdInput,
+  },
+  props: {
+    ...HdInput.props,
+  },
+  data() {
+    return {
+      currentValue: this.value,
+      currentType: 'password',
+    };
+  },
+  computed: {
+    showVisibilityToggle() {
+      return this.currentValue.length !== 0;
+    },
+  },
+  watch: {
+    value() {
+      this.currentValue = this.value;
+    },
+  },
```

Comment:
Correct me if I'm wrong again, please, but looks like you're doing it jus to bind to `<HdInput>` the `v-model` from `<HdInputPassword>`, right? Is it the case? Or there is something that I'm missing here?